src/main/Numbers.java

```diff
@@ -0,0 +1,40 @@
+package main;
+
+import java.util.Collections;
+import java.util.List;
+
+public class Numbers<T extends Number> {
+    private List<T> numbersList;
+
+    public Numbers(List<T> numbersList) {
+        this.numbersList = numbersList;
+    }
+
+    public int countEvenNumbers() {
+        int counter = 0;
+        for (T number : numbersList) {
+            if (Math.abs(number.intValue()) % 2 == 0) {
+                counter++;
+            }
+        }
+        return counter;
+    }
+
+    /**
+     * This method sets list 1's element with an index corresponding to index parameter
+     * to a value of an element of list 2 with the same index preventing the reverse operation.
+     * */
+    public void setValue(List<T> list1, final List<T> list2, int index) {
+        list1.set(index, list2.get(index));
```

Comment:
4. Create utility method, which takes two lists and index and replaces element at given position in the first collection with corresponding element from the second one, without possibility to do vice versa.

It shouldn't be possible to white `list2.set(index, list1.get(index));`. Can you try it?